examples/gke-a3ultragpu-staging.yaml

```diff
@@ -0,0 +1,181 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+blueprint_name: gke-a3-ultra-staging
+
+vars:
+  project_id: hpc-toolkit-dev-staging
+  deployment_name: gke-a3-ultra-staging
+  region: us-east5
+  zone: us-east5-staginga
+  # Cidr block containing the IP of the machine calling terraform.
+  # The following line must be updated for this example to work.
+  authorized_cidr: 10.88.0.4/32
+
+deployment_groups:
+- group: primary
+  modules:
+  - id: a3ultra-gke-net-0
+    source: modules/network/vpc
+    settings:
+      network_name: a3ultra-gke-net-0
+      allowed_ssh_ip_ranges: [0.0.0.0/0]
+      mtu: 8996
+      subnetworks:
+      - subnet_name: a3ultra-gke-sub-0
+        subnet_region: $(vars.region)
+        subnet_ip: 192.168.0.0/18
+      secondary_ranges:
+        a3ultra-gke-sub-0:
+        - range_name: pods
+          ip_cidr_range: 10.4.0.0/14
+        - range_name: services
+          ip_cidr_range: 10.0.32.0/20
+      firewall_rules:
+      - name: a3ultra-gke-internal-0
+        ranges: [192.168.0.0/16]
+        allow:
+        - protocol: tcp
+          ports: ["0-65535"]
+        - protocol: udp
+          ports: ["0-65535"]
+        - protocol: icmp
+
+  - id: a3ultra-gke-net-1
+    source: modules/network/vpc
+    settings:
+      network_name: a3ultra-gke-net-1
+      mtu: 8996
+      subnetworks:
+      - subnet_name: a3ultra-gke-sub-1
+        subnet_region: $(vars.region)
+        subnet_ip: 192.168.64.0/18
+      firewall_rules:
+      - name: a3ultra-gke-internal-1
+        ranges: [192.168.0.0/16]
+        allow:
+        - protocol: tcp
+          ports: ["0-65535"]
+        - protocol: udp
+          ports: ["0-65535"]
+        - protocol: icmp
+
+  - id: a3ultra-gke-rdma-net
+    source: community/modules/network/rdma-vpc
+    settings:
+      network_name: a3ultra-gke-rdma-net
+      network_profile: https://www.googleapis.com/compute/staging_alpha/projects/$(vars.project_id)/global/networkProfiles/titanium-mrdma
+      network_routing_mode: REGIONAL
+      subnetworks_template:
+        name_prefix: a3ultra-gke-mrdma-sub
+        count: 8
+        ip_range: 192.168.128.0/18
+        region: $(vars.region)
+
+  - id: a3-ultragpu-cluster
+    source: modules/scheduler/gke-cluster
+    use: [a3ultra-gke-net-0]
+    settings:
+      min_master_version: 1.30.4-gke.1348000
+      enable_private_endpoint: false  # Allows for access from authorized public IPs
+      rdma_subnetwork_name_prefix: $(a3ultra-gke-rdma-net.subnetwork_name_prefix)
+      is_a3_ultra: true
+      master_authorized_networks:
+      - cidr_block: $(vars.authorized_cidr)  # Allows your machine run kubectl command. It's required for the multi-network setup.
+        display_name: "kubectl-access-network"
+      additional_networks:
+        $(concat(
+          [{
+            network=a3ultra-gke-net-1.network_name,
```

Comment:
Quick question : Do we only need to pass in a single network, even though we created multiple networks ?

Reply:
All networks / subnetworks are being passed. `additional_networks` is a concatenation of this one network and the `a3ultra-gke-rdma-net.subnetwork_interfaces_gke`. The latter is a list of 8 items.